dist/tools/ci/print_toolchain_versions.sh

```diff
@@ -5,21 +5,17 @@ get_cmd_version() {
         return
     fi
 
-    local cmd="$1"
-    if command -v "$cmd" 2>&1 >/dev/null; then
-        ver=$("$cmd" --version 2> /dev/null | head -n 1)
-        # some tools (eg. openocd) print version info to stderr
-        if [ -z "$ver" ]; then
-            ver=$("$cmd" --version 2>&1 | head -n 1)
-        fi
-        if [ -z "$ver" ]; then
-            ver="error"
-        fi
-    else
-        ver="missing"
+    VERSION_RAW=$( ($@ --version) 2>&1)
+    ERR=$?
+    VERSION=$(echo "$VERSION_RAW" | head -n 1)
```

Comment:
it is more reliable to use `printf %s "$VERSION_RAW"` instead of echo to avoid getting strange outputs in certain situations, especially since `$VERSION_RAW` may contain anything.

Reply:
This has the `head` to make sure we get the fist line in commands that have more than one.